header/p2p/server.go

```diff
@@ -151,48 +159,108 @@ func (serv *ExchangeServer) requestHandler(stream network.Stream) {
 // if it exists.
 func (serv *ExchangeServer) handleRequestByHash(hash []byte) ([]*header.ExtendedHeader, error) {
 	log.Debugw("server: handling header request", "hash", tmbytes.HexBytes(hash).String())
-
 	ctx, cancel := context.WithTimeout(serv.ctx, serv.Params.ServeTimeout)
 	defer cancel()
+	ctx, span := serverTracer.Start(ctx, "ExchangeServer_handleRequestByHash", trace.WithAttributes(
+		attribute.String("hash", tmbytes.HexBytes(hash).String()),
+	))
+	defer span.End()
 
 	h, err := serv.getter.Get(ctx, hash)
 	if err != nil {
 		log.Errorw("server: getting header by hash", "hash", tmbytes.HexBytes(hash).String(), "err", err)
+		span.RecordError(err)
+		span.SetStatus(codes.Error, "")
 		return nil, err
 	}
+
+	span.AddEvent("fetched_header_from_store", trace.WithAttributes(
```

Comment:
same for events, would be great seeing them organized in consts

Reply:
also, not necessary.